CaptureRange.c

```diff
@@ -0,0 +1,59 @@
+#include "CaptureRange.h"
+#inlcude <stdlib.h>
+
+
+int CmpSamples(const void * Value1,const void * Value2)
+{
+   return (*Value1 - *Value2);
+}
+
+int CalcRange(int ContinuedCnt,int RangeCnt,int FirstSample,int LastSample)
+{
+  if(ContinuedCnt != 0)
+  {
+    RangeCnt++;
+    printf("Range,Readings\n");
+    printf("%d - %d, %d",FirstSample,LastSample,ContinuedCnt + 1);
+    RangeValue[RangeCnt][0] = FirstSample;
+    RangeValue[RangeCnt][1] = LastSample;
+    RangeValue[RangeCnt][2] = ContinuedCnt + 1;
+  }
+ return RangeCnt;
+}
+      
+int * DetectRange(int CurrentSamples[],int CurrentSamplesSize)
+{
+  int ContinuedCnt = 0;
+  int Consecutive = 0;
+  int RangeCnt = 0;
+  int FirstSample = CurrentSamples[0];
+  int LastSample = CurrentSamples[0];
+  int Index = 0;
+  int SampleDiff = 0;
+  
+  qsort(CurrentSamples,CurrentSamplesSize,sizeof(int),CmpSamples);
+  
+  for(Index=0;Index<CurrentSamplesSize;Index++)
+  {
+     SampleDiff = CurrentSamples[Index+1] - CurrentSamples[Index];
+     
+    if((SampleDiff == 0) || (SampleDiff == 1))
+    {
+       Consecutive = 1;
+    }
+    if(Consecutive == 1)
+    {
+       ContinuedCnt++;
+       LastSample = CurrentSamples[Index + 1];
+    }
+    else
+    {
+      RangeCnt = CalcRange(ContinuedCnt,RangeCnt,FirstSample,LastSample);
+      FirstSample = CurrentSamples[Index + 1];
+      LastSample =  CurrentSamples[Index];
+      ContinuedCnt = 0;
+    }
+  }
+  RangeValue[RangeCnt][3] = RangeCnt;
```

Comment:
A better structure should be used to send complex data instead of 2d array. That adds readability.

Reply:
ok.